qemu/tests/ioeventfd.py

```diff
@@ -0,0 +1,166 @@
+        """
+        ioevent_qtree_val = 'true' if 'on' in ioeventfd_opt else 'false'
+        logging.info('Execute info qtree in QMP monitor.')
+        qtree = qemu_qtree.QtreeContainer()
+        qtree.parse_info_qtree(vm.monitor.human_monitor_cmd("info qtree"))
+        for node in qtree.get_nodes():
+            if isinstance(node, qemu_qtree.QtreeDev) and (
+                    node.qtree.get('id', None) == dev_id):
+                if node.qtree.get('ioeventfd', None) is None:
+                    test.fail('The qtree device %s has no property ioeventfd.'
+                              % dev_id)
+                elif node.qtree['ioeventfd'] == ioevent_qtree_val:
+                    logging.info(
+                        'The \"%s\" matches with qtree device \"%s\"(%s).' %
+                        (ioeventfd_opt, dev_id, ioevent_qtree_val))
+                    break
+                else:
+                    test.fail(
+                        'The \"%s\" mismatches with qtree device \"%s\"(%s).' %
+                        (ioeventfd_opt, dev_id, ioevent_qtree_val))
+        else:
+            test.error('No such \"%s\" qtree device.' % dev_id)
+
+    def _get_ioeventfds(ioeventfd_opt):
+        """
+        Get the number of ioeventfds insid host.
+        """
+        logging.info('Check the \"%s\" via /proc/$PID/fd/.' % ioeventfd)
+        dst_log = 'off' if 'off' in ioeventfd_opt else 'on'
+        cmd = 'ls -l /proc/$(pgrep qemu-kvm)/fd > /tmp/%s' % dst_log
+        logging.debug('Running \'%s\'' % cmd)
+        s, o = process.getstatusoutput(cmd)
+        if s:
+            test.error('Failed to get the number of event fd.\n%s' % o)
+
+    def _compare_ioeventfds():
+        """
+        Compare the number of ioeventfds between "off" and "on".
+        """
+        logging.info('Compare the output of \'ls -l /proc/$PID/fd/\'.')
+        cmd = 'grep -c eventfd /tmp/off /tmp/on'
+        logging.debug('Running \'%s\'' % cmd)
+        s, o = process.getstatusoutput(cmd)
+        if s:
+            test.error('Failed to compare the outputs.\n%s' % s)
+        logging.debug(o)
+        nums = re.findall(r'\w+:(\d+)', o, re.M)
+        if int(nums[0]) > int(nums[1]):
+            test.fail('The number of event fds with \"off\" '
+                      'should be less than the one with \"on\".')
+        logging.info('The number of event fds with \"off\" '
+                     'is greater than the one with \"on\".')
+
+    params['start_vm'] = 'yes'
+    os_type = params['os_type']
+    ioeventfds = (params['orig_ioeventfd'], params['new_ioeventfd'])
+    for ioeventfd in ioeventfds:
+        dev_id = _set_ioeventfd_options()
+        logging.info('Boot a guest with \"%s\".' % ioeventfd)
+        env_process.preprocess_vm(test, params, env, params["main_vm"])
+        vm = env.get_vm(params["main_vm"])
+        vm.verify_alive()
+        session = vm.wait_for_login(timeout=float(params.get("login_timeout", 240)))
```

Comment:
I think you can use default `timeout` and in line 160, 161

Reply:
Thanks, will update.